drivers/clock_control/CMakeLists.txt

```diff
@@ -3,7 +3,7 @@
 zephyr_library()
 
 zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_BEETLE              beetle_clock_control.c)
-zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_CAVS                clock_control_cavs.c)
+zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_ADSP                clock_control_cavs.c)
```

Comment:
If we're going to change the Kconfig symbol, we should also change the relevant filenames to match.

Reply:
I'm just gonna go ahead and rename the functions too, while I'm at it (on a separate commit).

Reply:
Talked with Anas and he let me know that SOF would find it pretty disruptive to have a bunch of function name changes go in so I just left it to the filename changes (as you said) + some DTS-related changes. They're a commit on top - let me know what you think.

Reply:
It shouldn't be disruptive to change function names in the clock control driver, unless SOF is calling them directly instead of through the clock control API. But it's fine with me to leave them as-is.